com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -316,7 +316,7 @@ public AbstractMaterialNode outputNode
 
         public bool didActiveOutputNodeChange { get; set; }
 
-        internal delegate void SaveGraphDelegate(Shader shader);
+        internal delegate void SaveGraphDelegate(SaveContext shader);
```

Comment:
I'd rather have the API look like this, and then get rid of the base `SaveContext`. Then we remove the need for a HDRP/URP class inheriting a Shader Graph class.
```suggestion
        internal delegate void SaveGraphDelegate(Shader shader, object context);
```